### PR TITLE
Docs Redesign

### PR DESCRIPTION
**Pull request by joshreini1**
Changes include:
* Addition of top nav
* mkdocs navigation: indexes, instant, instant.prefetch, sections, sticky tabs, top, tracking, path
* mkdocs search: share & suggest
* mkdocs toc: follow
* header colors in trulens theme
* removal of unmaintainable how-to-use and other images of code
* Make docs the emphasized nav button
* Change main page header text from "experiments" to "applications"
* Add all quickstarts to the docs
* Clearly outline value props: eval + instrumentation
* Add more explainers on each value prop in separate sections
* Separate API refs into sections above
* Improve readability of smoke testing
* Some docstrings improvements

Note: need llama-index async PR to be merged before tests can pass.

### CODE REVIEW

File: docs/trulens_eval/trulens_eval_gh_top_readme.ipynb
Comment (rshih32) [inline docs/trulens_eval/trulens_eval_gh_top_readme.ipynb:1]: Juuust double checking; we are intentionally removing the code portions of the top level readme then?

Comment (joshreini1) [reply docs/trulens_eval/trulens_eval_gh_top_readme.ipynb:1]: Yes - intentionally removing the code portions of the toplevel readme.

File: .github/workflows/combine_nb_to_docs_testing.sh [hunk 1]
```diff
@@ -1,22 +1,18 @@
-rm -rf all_tools.ipynb
 rm -rf break.md
 
-# Combined notebook flow - will be tested
-nbmerge quickstart.ipynb logging.ipynb feedback_functions.ipynb >> all_tools.ipynb
```
Comment (rshih32) [inline .github/workflows/combine_nb_to_docs_testing.sh:5]: Don't remove this; it still needs to be tested

File: .github/workflows/combine_nb_to_docs_testing.sh [hunk 2]
```diff
@@ -1,22 +1,18 @@
...
-
 # Colab quickstarts
 nbmerge colab_dependencies.ipynb langchain_quickstart.ipynb >> langchain_quickstart_colab.ipynb
 nbmerge colab_dependencies.ipynb llama_index_quickstart.ipynb >> llama_index_quickstart_colab.ipynb
 nbmerge colab_dependencies.ipynb no_framework_quickstart.ipynb >> no_framework_quickstart_colab.ipynb
 
 # Create pypi page documentation
-jupyter nbconvert --to markdown all_tools.ipynb
-printf  "\n\n" >> break.md
-cat intro.md break.md all_tools.md > README.md
```
Comment (rshih32) [inline .github/workflows/combine_nb_to_docs_testing.sh:15]: double checking again; now we are removing all code from the trulens_eval/README as well?

File: .github/workflows/combine_nb_to_docs_testing.sh [hunk 3]
```diff
@@ -1,22 +1,18 @@
-rm -rf all_tools.ipynb
```
Comment (rshih32) [inline .github/workflows/combine_nb_to_docs_testing.sh:1]: keep this too

File: .github/workflows/combine_nb_to_docs_testing.sh [hunk 4]
```diff
@@ -31,22 +27,21 @@ SED=`which -a gsed sed | head -n1`
...
 # Move all generated files to the generated_files folder
 mv README.md ../../trulens_eval/README.md
 mv TOP_README.md ../../README.md
 
+mv *langchain_quickstart_colab.ipynb ../../trulens_eval/examples/colab/quickstarts/
 mv llama_index_quickstart.py ../../trulens_eval/examples/llama_index_quickstart.py
 mv no_framework_quickstart.py ../../trulens_eval/examples/no_framework_quickstart.py
-mv *.py ../../trulens_eval/examples/
-mv *quickstart_colab.ipynb ../../trulens_eval/examples/colab/quickstarts/
-mv all_tools* ../../trulens_eval/generated_files/
```
Comment (rshih32) [inline .github/workflows/combine_nb_to_docs_testing.sh:52]: why are these removed? This is does our testing, colab notebooks, and non-colab examples

File: docs/trulens_eval/intro.md [hunk 5]
```diff
@@ -6,58 +6,56 @@ Evaluate and track your LLM experiments with TruLens. As you work on your models
...
-Our example chat app and feedback functions call external APIs such as OpenAI or HuggingFace. You can add keys by setting the environment variables. 
-
-#### In Python
-
-```python
-import os
-os.environ["OPENAI_API_KEY"] = "..."
-```
-
-#### In Terminal
```
Comment (rshih32) [inline docs/trulens_eval/intro.md:59]: Why do we remove this? Isn't this the key instructions as to how to get started?

**Review by rshih32 — CHANGES_REQUESTED** [review]

File: .github/workflows/combine_nb_to_docs_testing.sh (hunk 4 above)
Comment (joshreini1) [reply .github/workflows/combine_nb_to_docs_testing.sh]: Undid these changes

File: docs/trulens_eval/intro.md (hunk 5 above)
Comment (joshreini1) [reply docs/trulens_eval/intro.md:59]: Replaced the code with links to 3 different quickstarts to emphasize TruLens is cross-framework.

File: .github/workflows/combine_nb_to_docs_testing.sh (hunk 2 above)
Comment (joshreini1) [reply .github/workflows/combine_nb_to_docs_testing.sh:15]: correct, replacing with links to all quickstarts. Emphasizing that TruLens is cross-framework.

File: trulens_eval/trulens_eval/tests/qs_relevance_smoke_tests.ipynb [hunk 6]
```diff
@@ -528,11 +518,11 @@
       "+---------------------------+-------------+-------------+---------------------+\n",
       "| Test Type                 |   Passed ✅ |   Failed ❌ | Percentage Passed   |\n",
       "+===========================+=============+=============+=====================+\n",
-      "| Increasingrelevance Tests |           3 |           0 | 100.00%             |\n",
```
Comment (shayaks) [inline trulens_eval/trulens_eval/tests/qs_relevance_smoke_tests.ipynb:531]: Looks like we regressed a lot in our tests.

**Review by shayaks — APPROVED** [review]

File: docs/trulens_eval/intro.md [hunk 7]
```diff
@@ -6,58 +6,56 @@ Evaluate and track your LLM experiments with TruLens. As you work on your models
...
+### 💡 Contributing
 
-```bash
-export OPENAI_API_KEY = "..."
-```
```
Comment (rshih32) [inline docs/trulens_eval/intro.md:63]: lets keep this

**Review by rshih32 — CHANGES_REQUESTED** [review]

**Review by rshih32 — APPROVED** [review]
